src/ngx_pagespeed.cc

```diff
@@ -422,6 +422,20 @@ ngx_int_t ps_set_cache_control(ngx_http_request_t* r, char* cache_control) {
   return NGX_OK;
 }
 
+StringPiece ps_get_cache_control(ngx_http_request_t* r) {
+  ngx_table_elt_t* header;
+  NgxListIterator it(&(r->headers_out.headers.part));
+  while ((header = it.Next()) != NULL) {
+    if (STR_CASE_EQ_LITERAL(header->key, "Cache-Control")) {
```

Comment:
I'm wondering if we should also take the `cache_control` field of `ngx_http_headers_out_t` [1] into account?

http://lxr.nginx.org/source/src/http/ngx_http_request.h#0271